notebook2/notebook/notebook.py

```diff
@@ -473,27 +421,34 @@ def wait_websocket(ws):
     if notebook is None:
         return
 
-    pod_name = notebook['pod_name']
-    svc_name = notebook['svc_name']
-    while True:
+    pod_uuid = notebook['pod_uuid']
+    url = external_url_for(f'instance-ready/{pod_uuid}/')
+
+    attempts = -1
+    while attempts < 10:
+        attempts += 1
+        # Protect against many requests being issued
+        # This may happen if client re-reqeusts after seeing 405
+        # which can occur if it relies on container status,
+        # which may not be "Ready" immediately after jupyter server is reachable
+        # https://youtu.be/dp-RYuH4tmw
+        gevent.sleep(1)
+
         try:
-            response = requests.head(f'https://notebook2.hail.is/instance-ready/{svc_name}/',
-                                     timeout=1)
-            if response.status_code < 500:
-                log.info(f'HEAD on jupyter succeeded for {svc_name} {pod_name} response: {response}')
-                # if someone responds with a 2xx, 3xx, or 4xx, the notebook
-                # server is alive and functioning properly (in particular, our
-                # HEAD request will return 405 METHOD NOT ALLOWED)
-                break
+            response = requests.head(url, timeout=1, cookies=request.cookies)
+
+            if response.status_code == 502:
+                log.info(f'Pod not reachable for pod_uuid: {pod_uuid} : response: {response}')
```

Comment:
All other comments removed as the log.info message seems to clearly communicate that information